src/tribol/search/InterfacePairFinder.cpp

```diff
@@ -509,16 +514,21 @@ class GridSearch
        {
           // Preliminary geometry/proximity checks, SRW
           InterfacePair pair( meshId1, cellType1, fromIdx,
-                              meshId2, cellType2, toIdx, false,
+                              meshId2, cellType2, toIdx, true,
                               -1 );
-          bool contact = geomFilter( pair, cs->getContactMode() );
+          bool isContactCandidate = geomFilter( pair, cs->getContactMode() );
 
-          if (contact)
+          if (isContactCandidate)
           {
+             pair.isContactCandidate = true;
              pair.pairId = k;
              contactPairs->addInterfacePair( pair );
              ++k;
           }
+          else
+          {
+             pair.isContactCandidate = false;
+          }
```

Comment:
Seems like this block is pretty similar to the above one (L435-449).  Can they be moved to a single function?  Maybe a new InterfacePair constructor?

Reply:
We have to check the return value of the geomFilter routine to determine whether to add the interface pair to the manager or not. I moved setting pair.isContactCandidate inside geomFilter (since geomFilter is in fact responsible for performing initial checks on valid face-pairs), which simplifies the lines you point out here. I think this is cleaner?